code-collab-app/src/homepage.js

```diff
@@ -3,49 +3,109 @@ import logo from './codeCollabLogo.png';
 import './App.css';
 import { Button, ListGroup } from 'react-bootstrap';
 import { Link } from 'react-router-dom';
+import peer_data from './storage/peer_data';
 
-function Homepage() {
-
-  return (
-    <div className="App">
-      <header className="App-header">
-        <span>
-          <Link to='/'><img src={logo} className="App-logo" alt="logo"/></Link>{'  '}
-          <Link to='/editor'><Button variant='dark'>New Document</Button></Link>{'  '}
-          {/* TODO: Add upload functionality and redirect to editor page */}
-          <Button variant='dark'>Upload Document</Button>
-        </span>
-      </header>
-      <body>
-        {/* TODO: Dynamically update this with active sessions */}
-        <div className="container">
-          <h1 className='subheading'>Active Sessions</h1>
-          <div className="row">
-            <div className="col-sm">
-              <ListGroup.Item>Session 1</ListGroup.Item>
-            </div>
-            <div className="col-sm">
-              <ListGroup.Item>Session 2</ListGroup.Item>
-            </div>
-            <div className="col-sm">
-              <ListGroup.Item>Session 3</ListGroup.Item>
-            </div>
-          </div>
-          <div className="row">
-            <div className="col-sm">
-              <ListGroup.Item>Session 4</ListGroup.Item>
-            </div>
-            <div className="col-sm">
-              <ListGroup.Item>Session 5</ListGroup.Item>
-            </div>
-            <div className="col-sm">
-              <ListGroup.Item>Session 6</ListGroup.Item>
-            </div>
-          </div>
+class Homepage extends React.Component {
+  constructor(props){
+    super(props);
+    
+    peer_data.initialize();
+    this.activeSessions = peer_data.session_list;
+  }
+
+  handleJoinSession(session_info) {
+    peer_data.join_session(session_info[0], session_info[1]);
+    this.props.history.push({
+      pathname: '/editor',
+      data: peer_data.doc_data
```

Comment:
@paulchenpmc this is the document data I will be passing from the from homepage to the editor page when the user clicks on an active session. So this data should be what needs to be formatted on the editor page.

Reply:
Sounds g